theano/sandbox/rng_mrg.py

```diff
@@ -13,17 +13,15 @@
 from six.moves import xrange
 
 import theano
-from theano import Op, Apply, shared, config, Variable
-from theano import gradient, function
-from theano.gradient import undefined_grad
+from theano import Op, Apply
 from theano import tensor
-from theano.tensor import (TensorType, as_tensor_variable, get_vector_length,
-                           cast, opt, scal)
-from theano.tensor import sqrt, log, sin, cos, join, prod
-from theano.compile import optdb
-from theano.gof import local_optimizer, ParamsType
+from theano.tensor import (TensorType, as_tensor_variable, scal)
+from theano.gradient import grad_undefined, undefined_grad
 from theano.scalar import bool as bool_t, int32 as int_t
-from . import multinomial
+from theano.sandbox import multinomial
+
+floatX = theano.config.floatX
+intX = floatX.replace('float', 'int')
```

Comment:
I don't like this change. This change the user interface just before a release and isn't good. If you want this PR to be included in the next release, can you keep in it just the changes about the new random generator?

Also, did you see the announce: https://groups.google.com/d/msg/theano-users/7Poq8BZutbY/rNCIfvAEAwAJ

This mostly tell we won't change the interface, so moving this interface change to another PR isn't useful.

Reply:
Would it be okay to use `_floatX` and `_intX` instead to make them *private*. That would leave the interface unchanged (assuming that private variables are not part of the interface).

Note: as you might notice, I am not a full-blood python developer...

Reply:
This isn't the interface change that cause problem. The problem is that when you set floatX=float32, then the dtype change compared to before. In the past, but default it was always int64 and didn't looked at floatX. It should stay like that.

Reply:
@nouiz I see, but wouldn't it make more sense to have `int32` as default when `floatX` is `float32` and `int64` when `floatX` is `float64`? E.g. for people working on 32-bit hardware...

Reply:
float32 isn't a questions of 32 bit hardware or not. You can want to use float32, but need int64 for indexing. There was a discussion to make an intX in the past. I forgot why we didn't do it, probably not enough time to get to it. But now isn't a good time to implement that.